nums/core/array/blockarray.py

```diff
@@ -883,12 +883,26 @@ def _block_shape_reshape(self, arr, block_shape):
     def _strip_ones(self, shape):
         return tuple(filter(lambda x: x != 1, shape))
 
+    def _check_positions_ones(self, shape, block_shape):
+        for i in range(len(shape)):
+            if shape[i] == 1:
+                if shape[i] != block_shape[i]:
+                    return False
+        return True
+
     def _is_simple_reshape(self, arr: BlockArray, shape, block_shape):
         # Is the reshape a difference of factors of 1?
         # Strip out 1s and compare.
```

Comment:
Update this comment to reflect the increased strictness of our check.

Reply:
Done